lib/rules/comma-dangle.js

```diff
@@ -81,49 +81,49 @@ module.exports = {
             category: "Stylistic Issues",
             recommended: false
         },
-
         fixable: "code",
-
-        schema: [
-            {
-                defs: {
-                    value: {
-                        enum: [
-                            "always",
-                            "always-multiline",
-                            "only-multiline",
-                            "never"
-                        ]
-                    },
-                    valueWithIgnore: {
-                        anyOf: [
-                            {
-                                $ref: "#/defs/value"
```

Comment:
@not-an-aardvark , now I understand the problem here. When you pass array as a rule schema (`schema: [obj]`, where obj as a JSON schema) eslint internally converts it to a following JSON schema:

```javascript
{
  type: 'array',
  items: [
    obj
  ],
  minItems: 0,
  maxItems: 1
}
```

If there are definitions inside obj (like it is with comma-dangle) then their JSON pointers should be extended as well, otherwise they are not found. is-my-json-valid quietly ignores $ref's that are not found and ajv (by default) throws exception. This change to the schema fixes this issue. In addition there are "$refs" instead of "$ref" in several places, that ignored by both is-my-json-valid and by ajv.

There may be some third party rules with similar incorrect usage of definitions not on the top level but inside item schema and these rules will start logging errors. Should it be a breaking change? I am not sure. These rules are not working correctly now so maybe it is ok if they have to be fixed. There are the following options:

1. Leave things as they are (therefore breaking third-party rules with incorrect schemas)
2. Change Ajv settings to log warnings when $ref's are not found instead of throwing exception, so things will work as they do now and all rules can be fixed separately (including comma-dangle.
3. Keep things completely as they are, quietly ignoring $ref's that cannot be found.
4. Improve eslint to correctly extend JSON pointers in the above scenario, so internally "#/defs/value" would become "#/items/0/defs/value". That's my least favourite option, as it can be confusing to debug and some third-party rules can still start breaking as currently they simply don't validate against these schemas and then they start, so if something was invalid it would start breaking.

I think option 2 is the safest, but let me know what you think.